ComponentKitApplicationTests/CKComponentBoundsAnimationTests.mm

```diff
@@ -0,0 +1,145 @@
+
+  const CKBuildComponentResult firstResult = CKBuildComponent([CKComponentScopeRoot rootWithListener:nil], {}, ^{
+    return [CKStackLayoutComponent
+            newWithView:{}
+            size:{}
+            style:{.alignItems = CKStackLayoutAlignItemsStretch}
+            children:{
+              {[CKBoundsAnimationComponent newWithIdentifier:@0], .flexGrow = YES},
+            }];
+  });
+  const CKComponentLayout firstLayout = [firstResult.component layoutThatFits:{{100, 100}, {100, 100}} parentSize:{}];
+  NSSet *firstMountedComponents = CKMountComponentLayout(firstLayout, container);
+
+  const CKBuildComponentResult secondResult = CKBuildComponent(firstResult.scopeRoot, {}, ^{
+    return [CKStackLayoutComponent
+            newWithView:{}
+            size:{}
+            style:{.alignItems = CKStackLayoutAlignItemsStretch}
+            children:{
+              {[CKBoundsAnimationComponent newWithIdentifier:@0], .flexGrow = YES},
+              {[CKBoundsAnimationComponent newWithIdentifier:@1], .flexGrow = YES},
+            }];
+  });
+  const CKComponentLayout secondLayout = [secondResult.component layoutThatFits:{{100, 100}, {100, 100}} parentSize:{}];
+  NSSet *secondMountedComponents = CKMountComponentLayout(secondLayout, container);
+
+  XCTAssertEqual([container.subviews count], 2u);
+  XCTAssertTrue(((CKBoundsAnimationRecordingView *)container.subviews[0]).animatedLastBoundsChange);
+  XCTAssertFalse(((CKBoundsAnimationRecordingView *)container.subviews[1]).animatedLastBoundsChange);
+
+  [firstMountedComponents makeObjectsPerformSelector:@selector(unmount)];
+  [secondMountedComponents makeObjectsPerformSelector:@selector(unmount)];
+}
+
+- (void)testBoundsAnimationIsNotAppliedWhenViewRecycledForComponentWithDistinctScopeFrameToken
+{
+  UIView *container = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
+
+  const CKBuildComponentResult firstResult = CKBuildComponent([CKComponentScopeRoot rootWithListener:nil], {}, ^{
+    return [CKBoundsAnimationComponent newWithIdentifier:@0];
+  });
+  const CKComponentLayout firstLayout = [firstResult.component layoutThatFits:{{50, 50}, {50, 50}} parentSize:{}];
+  NSSet *firstMountedComponents = CKMountComponentLayout(firstLayout, container);
+
+  const CKBuildComponentResult secondResult = CKBuildComponent(firstResult.scopeRoot, {}, ^{
+    return [CKBoundsAnimationComponent newWithIdentifier:@1];
+  });
+  const CKComponentLayout secondLayout = [secondResult.component layoutThatFits:{{100, 100}, {100, 100}} parentSize:{}];
+  NSSet *secondMountedComponents = CKMountComponentLayout(secondLayout, container);
+
+  CKBoundsAnimationRecordingView *v = (CKBoundsAnimationRecordingView *)secondResult.component.viewContext.view;
+  XCTAssertFalse(v.animatedLastBoundsChange);
+
+  [firstMountedComponents makeObjectsPerformSelector:@selector(unmount)];
+  [secondMountedComponents makeObjectsPerformSelector:@selector(unmount)];
+}
+
+@end
```

Comment:
Are there no tests that actions are re-enabled once the mount animation guard goes out of scope? Should there be? Disabling the actions affects the global state, right?